cogs/wordfilter/wordfilter.py

```diff
@@ -550,20 +570,100 @@ async def on_message(self, msg):
+                    title=f"Filtered words for **{ctx.guild.name}**", description=page
+                )
+                embed.set_footer(text=f"Page {pageNumber}/{totalPages} ({totalEntries} entries)")
+                embed.colour = discord.Colour.red()
+                pageList.append(embed)
+            await menu(ctx, pageList, DEFAULT_CONTROLS)
+        else:
+            await user.send("Sorry you have no filtered words in **{}**".format(ctx.guild.name))
+
+    @stat.command(name="orderedusage")
+    @commands.guild_only()
+    @checks.mod_or_permissions(manage_messages=True)
+    async def orderedCensorUsageList(self, ctx):
+        """
+        displays an ordered usage list of all the censored words that have been used
+        """
+        user = ctx.message.author
+        rawUsageStats = await self.config.guild(ctx.guild).get_attr(KEY_USAGE_STATS)()
+
+        if rawUsageStats:
+            display = []
+            pageList = []
+            count = 1
+            for regex, timesUsed in dict(
+                reversed(sorted(rawUsageStats.items(), key=lambda item: item[1]))
+            ).items():
+                display.append(f"{count}. `{regex}` : `{timesUsed}`,")
+                count += 1
+            msg = "\n".join(display)
+            pages = list(chat_formatting.pagify(msg, page_length=400))
+            totalPages = len(pages)
+            totalEntries = len(display)
+
+            async for pageNumber, page in AsyncIter(pages).enumerate(start=1):
+                embed = discord.Embed(
+                    title=f"Filtered words for **{ctx.guild.name}**", description=page
+                )
+                embed.set_footer(text=f"Page {pageNumber}/{totalPages} ({totalEntries} entries)")
+                embed.colour = discord.Colour.red()
+                pageList.append(embed)
+            await menu(ctx, pageList, DEFAULT_CONTROLS)
+        else:
+            await user.send("Sorry you have no filtered words in **{}**".format(ctx.guild.name))
+
 
 def _censorMatch(matchobj):
     matchLength = len(matchobj.group(0))
     return "`" + ("*" * matchLength) + "`"
 
 
 def _filterWord(words, string):
-    # Combine all filter words into one regex
-    numFilters = len(words) - 1
-    reFormat = r"\b(?:" + (r"{}|") * numFilters + r"{})\b"
-    regex = reFormat.format(*words)
-
-    # Replace the offending string with the correct number of stars.
-    return re.sub(regex, _censorMatch, string, flags=re.IGNORECASE)
+    numWords = len(words)
+    if not words:
+        # if no filters added yet, do nothing
+        return string
+    else:
+        # Combine all filter words into one regex
+        numFilters = numWords - 1
+        reFormat = r"\b(?:" + (r"{}|") * numFilters + r"{})\b"
+        regex = reFormat.format(*words)
+
+        # sees how many times each filtered word appears in string and tallies them up in the config file
+        for word in words:
+            wordInstances = len(re.findall(regex, word))
+
```

Comment:
Are we doing anything with this?

If this is from our sideband discussion earlier, we could probably move the stuff you were doing outside of this helper function, and convert this into a method inside the class in order to reference `self`.

Reply:
I just removed it. Because half of the time this command is being used is when it is checking if a message is already filtered, to add the counter in here would mess up the statistics